domains/measurements/Measurements.API/API/Services/ConsumptionAggregation.cs

```diff
@@ -0,0 +1,54 @@
+using API.Models;
+using EnergyOriginDateTimeExtension;
+using System.Linq;
+
+namespace API.Services;
+
+class ConsumptionAggregation : IConsumptionAggregator
+{
+    public MeasurementResponse CalculateAggregation(IEnumerable<TimeSeries> measurements, long dateFrom, long dateTo, Aggregation aggregation)
+    {
+        var listOfConsumptions = measurements.SelectMany(
+                                  measurement => measurement.Measurements.Select(
+                                  reading => new AggregatedMeasurementInteral{
+                                      DateFrom = reading.DateFrom.ToDateTime(),
+                                      DateTo = reading.DateTo.ToDateTime(),
+                                      Value = reading.Quantity
+                                  })).ToList();
+
+        IEnumerable<IGrouping<string, AggregatedMeasurementInteral>> groupedConsumptions = GetGroupedConsumption(aggregation, listOfConsumptions);
+
+        var bucketConsumptions = (from groupedConsumption in groupedConsumptions
+                                  let totalForBucket = groupedConsumption.Sum(_ => _.Value)
+                                  select new AggregatedMeasurement(
+                                        groupedConsumption.First().DateFrom.ToUnixTime(),
+                                        groupedConsumption.Last().DateTo.ToUnixTime(),
+                                        totalForBucket
+                                    )).ToList();

```

Comment:
There are still some SQL-style LINQ left over. Maybe take an extra look in regards to indentation and new lines too :D